src/firmware_new/boards/frankie_v1/main.c

```diff
@@ -129,17 +109,21 @@ int main(void)
 
     /* Initialize all configured peripherals */
     MX_GPIO_Init();
-    MX_ETH_Init();
     MX_USART3_UART_Init();
     MX_USB_OTG_FS_PCD_Init();
+    MX_CRC_Init();
+    MX_LWIP_Init();
     /* USER CODE BEGIN 2 */
+    HAL_Delay(1000);
+    udp_echoserver_init();
 
     /* USER CODE END 2 */
 
     /* Infinite loop */
     /* USER CODE BEGIN WHILE */
     while (1)
     {
+        MX_LWIP_Process();
```

Comment:
Just in case other people are using this project to do stuff, will this block and prevent other stuff from running? 

Also just so I can get a feel for where this is going, would this run in a RTOS task? At what frequency if so? Or would it be interrupt driven?

Reply:
nope it returns after its processed one frame IIRC, it defn doesn't block forever. 

just checked some RTOS examples, it seems like they are using netconn, and it looks like its just a thread w/ an infinite loop